File: php/primes/Simple.php
#!/usr/bin/env php
<?php

const NUMBER = 500000;

function getLastPrime($count) {
    $lastPrime = 2;

    // Traverse each number from 3 to N, skipping even numbers
    for ($i = 3; $i <= $count; $i += 2) {
        $isPrime = true;
        $limit = floor(sqrt($i));

        for ($j = 3; $j <= $limit; $j += 2) {
            if ($i % $j == 0) {
                $isPrime = false;
                break;
            }
        }

        if ($isPrime) {
            $lastPrime = $i;
        }
    }

    return $lastPrime;
}

(function () {
    $startTimeMs = floor(microtime(true) * 1000);

    $lastPrime = getLastPrime(NUMBER);
    echo "Last prime: " . $lastPrime . "\n";

    $endTimeMs = floor(microtime(true) * 1000);
    $durationMs = $endTimeMs - $startTimeMs;
  
    echo "Execution time: " . $durationMs . "ms\n";
})();

File: lua/primes/Simple.lua
#!/usr/bin/env lua

local NUMBER = 1000000

local function getLastPrime(count)
    local lastPrime = 2

    -- Traverse each number from 3 to N, skipping even numbers
    for i = 3, count, 2 do
        local isPrime = true
        local limit = math.floor(math.sqrt(i))

        for j = 3, limit, 2 do
            if i % j == 0 then
                isPrime = false
                break
            end
        end

        if isPrime then
            lastPrime = i
        end
    end

    return lastPrime
end

local function main()
    local startTime = os.clock()

    local lastPrime = getLastPrime(NUMBER)
    print("Last prime: " .. lastPrime)

    local endTime = os.clock()
    local durationMs = (endTime - startTime) * 1000

    print("Execution time: " .. durationMs .. "ms")
end

main()
